crates/primitives/src/precompile.rs

```diff
@@ -10,6 +10,33 @@ pub type PrecompileResult = Result<(u64, Bytes), PrecompileError>;
 pub type StandardPrecompileFn = fn(&Bytes, u64) -> PrecompileResult;
 pub type EnvPrecompileFn = fn(&Bytes, u64, env: &Env) -> PrecompileResult;
 
+/// Clonable precompile trait. It is used to create a boxed precompile.
+pub trait ClonablePrecompileTrait: DynClone + Send + Sync {
+    fn call(&self, bytes: &Bytes, gas_price: u64, env: &Env) -> PrecompileResult;
+}
+
+dyn_clone::clone_trait_object!(ClonablePrecompileTrait);
+
+/// Box over clonable precompile trait.
+pub type BoxedPrecompileTrait = Box<dyn ClonablePrecompileTrait>;
```

Comment:
is this just `Arc<dyn>`?

bu dynclone also works, it's probably nicer for no_std? idk

Reply:
It looks a lot like `Arc`, will need to check this statement but Box could allow mutability, this could be interesting. In this case we could even use arc.

Reply:
I think it doesn't matter, I envision this will be used like

```rust
#[derive(Clone)]
struct MyPrecompile {
  inner: Arc<Inner>
}
```

but I'd lean towards `Arc<dyn>`

Reply:
Added both of them, `StatefulPrecompile` as `Arc` and `StatefulPrecompileMut` as `Box`